### DISCUSSION TITLE
Odyssey, UI and 'Ship'

### DISCUSSION THREAD

**Discussion by Athanasius** [EDCD/EDMarketConnector#959] [General]
Today I've been working on making EDMC recognise when you're on-foot so as to allow things like 'Update' to work without errors.

Currently there are still limitations as the only time it can recognise you're on-foot at a station is if you login to the game whilst on-foot at one.  The 'Disembark' event gives no location data, and there's no 'Docked' or 'Location' even if you disembark from an Apex shuttle at one.

Anyway... now that's working I was going to work on changing the 'Ship: Sidewinder' line in the UI into something that will let you know what type of suit you're wearing when on-foot, and potentially (if either coriolis/edsy or another site is created for it) a link to examine and play with your suit loadout.

However some are of the opinion that it's fine as is, even if, as now with the Alpha Phase 1, you can't even use the ship it lists.

So, anyone else have opinions on this ?

**Comment by A-UNDERSCORE-D** [comment]
I still say we should show both suit and ship. If the player has no ship we can drop the line, and if the player is wearing a flightsuit we can show that too. I dont like the idea of dropping the ship you're in simply because _right now_ you're on foot.

**Reply by Athanasius** [reply]
Would it be confusing to know you have a ship you can use (when Odyssey Alpha progresses, or it's released), but be in a taxi and still see 'Ship: your ship name' ?

**Reply by A-UNDERSCORE-D** [reply]
I dont believe so, its somewhat a given that the taxi isnt yours

**Comment by Athanasius** [comment]
I *am* now coming around to keeping `Ship: ...` as-is (even if it's showing an out of reach ship for you, and you're in an Apex taxi at the moment), and adding a new `Suit: ...` line if detected on loadgame or otherwise (i.e. no point showing it to players without Odyssey).

For the Suit it would likely just show either your suit's type name (i.e. Maverick) *or* the active Loadout name if we have that information.

**Reply by anthonylangsworth** [reply]
This makes the fewest assumptions (Occam's Razor) and so sounds like a better solution to me.

**Reply by A-UNDERSCORE-D** [reply]
Agreed, gives you all the info you want / may need at once.

**Comment by Athanasius** [comment]
OK, so let's conclude we'll always keep 'Ship' and have an extra 'Suit' line.

What goes in the text to right right of the label?

1. Suit type/name.
2. Loadout name.
3. Both ?

I'm inclined for it to be both:

`Suit:   Maverick (Loadout 1)`

**Reply by A-UNDERSCORE-D** [reply]
both. use `/` to be consistent with `shipname / group owner`?

**Reply by Athanasius** [reply]
I was actively **not** using that same format, because it's really not the same semantics.

```
Cmdr: <your name> / <private group or multi-crew other-cmdr name>
Suit: <suit name> (<its loadout>)
```
The first is 'where you are', the second is 'and the suit is on this loadout'.

**Comment by Athanasius** [comment]
If anyone missed it ... the code for this is in 5.0.0-beta6 now.

**Comment by A-UNDERSCORE-D** [comment]
do we want to close /  lock this discussion?

**Comment by Athanasius** [comment]
Well, I've had more thoughts.

In lieu of a coriolis/edsy-like site to have links point to, how about seeing if we can make clicking a suit/loadout copy relevant JSON data ?

**Reply by A-UNDERSCORE-D** [reply]
that would work well. hell we may end up being the first to define a format
